easybuild/easyconfigs/n/NVHPC/NVHPC-20.7-GCC-9.3.0.eb

```diff
@@ -0,0 +1,51 @@
+name = 'NVHPC'
+version = '20.7'
+local_gccver = '9.3.0'
+versionsuffix = '-GCC-%s' % local_gccver
+
+homepage = 'https://developer.nvidia.com/hpc-sdk/'
+description = """C, C++ and Fortran compilers included with the NVIDIA HPC SDK (previously: PGI)"""
+
+toolchain = SYSTEM
+
+# NVHPC can be downloaded freely from NVIDIA's website at https://developer.nvidia.com/hpc-sdk
+# It requires accepting the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
+# If you accept the License Agreement, you may also uncomment the following line to automatically download the sources
+# source_urls = ['https://developer.download.nvidia.com/hpc-sdk/%(version)s/']
+sources = ['nvhpc_2020_%(version_major)s%(version_minor)s_Linux_x86_64_cuda_multi.tar.gz']
+checksums = ['a5c5c8726d2210f2310a852c6d6e03c9ef8c75e3643e9c94e24909f5e9c2ea7a']
+
+dependencies = [
+    ('GCCcore', local_gccver),
+    ('binutils', '2.34', '', ('GCCcore', local_gccver)),
+    ('CUDA', '11.0.2', '', ('GCC', local_gccver)),
```

Comment:
I think it would be better to use CUDAcore here. The GCC dep is going to mess up in an HMNS I think.

`('CUDAcore', '11.0.2'),`

Reply:
Hmm, yeah, good point actually... I overlooked `CUDAcore`.

@AndiH @damianam Thoughts?

Reply:
I don't mind.

From the easyconfig I see that `CUDACore` also uses the `CUDA` EasyBlock, right? In that case, `$CUDA_HOME` should also be set, I assume. That's the only hard necessity here.

One caveat, though: CUDA is optional for th NVHPC package. NVHPC is shipped with its own CUDA, which is taken when no external CUDA module (setting `$CUDA_HOME`) is there. If one would want to rather use the NVHPC-shipped CUDA, they would need to move from `CUDAcore` to `GCCcore`. That might be a bit … indirect? What do you think?

Reply:
With that in mind, I guess it makes more sense to just not include a specific `CUDA(core)` as a dep here? @bartoldeman?

Reply:
That would possibly be in line with what NVIDIA recommends doing.

For JSC, we (well… I…) decided that we want to use our dedicated CUDA module (which should be compatible as long as a proper version is picked and `$CUDA_HOME` is set), to have no hidden dependencies and use the other stack which is dependent on CUDA – and be more flexible in updating the CUDA version.

Reply:
@AndiH Let's go forward with not including a specific `CUDA` dependency then, so we can get this PR merged (and included in the upcoming EasyBuild release).

We've made sure in the easyblock that things work properly when you do include a particular CUDA as dependency, but the default should probably to go with what's included with NVHPC itself.

Reply:
Ok.
I'd add this to the block of comments!

Reply:
How do you like f7cc517a9b?

Reply:
agreed, if NVHPC (like PGI before which also included some CUDA) can be flexibly be combined with any CUDA module and is usable for compiling CPU-only software without CUDA_HOME set, than it should not be a runtime dependency.

Reply:
In the end I believe that because currently the easyconfig does not install out-of-the-box without `--try-amend=default_cuda_version=<version>` it should use the version that the installer picks by default (11.0).

One can make that even more explicit using a versionsuffix, i.e.
versionsuffix = '-CUDA-11.0.2'
and then rename to:
NVHPC-20.7-GCC-9.3.0-CUDA-11.0.2.eb

Then using:
```dependencies = [('CUDAcore', '11.0.2')]```
locks the dependency into place and makes the behaviour of NVHPC consistent with or without a `CUDA` module loaded.. this consistency is needed since e.g. `nvc++ -stdpar` ignores CUDA_HOME (and then the only purpose of the `CUDA` module, unlike `CUDAcore`, is to just extend MODULEPATH).

Also note that if NVHPC is installed with the bundled CUDA, then CUDA_HOME is NOT set which in turn makes other software believe that CUDA is not installed, so I would keep the CUDAcore installation in any case; setting CUDA_HOME in the NVHPC module to the bundled CUDA would be an alternative but quite unusual.

but please delete the `dependencies = [('CUDA', '11.0.2', '', ('GCC', local_gccver))]` example since that won't work in an HMNS.

So here's my proposal:
1. add `versionsuffix = '-CUDA-11.0.2'`
2. rename to `NVHPC-20.7-GCC-9.3.0-CUDA-11.0.2.eb`
3. use `dependencies = [('CUDAcore', '11.0.2')]`
4. remove comment about `dependencies = [('CUDA', '11.0.2', '', ('GCC', local_gccver))]`

Reply:
I'll let @boegel vote on whether or not to use versionsuffix -- it's conceivable but not super likely to also have
`NVHPC-20.7-GCC-9.3.0-CUDA-10.2.89.eb`

Reply:
I'm fine with all of that and would implement it as soon as @boegel also gives his ok!

Should I create a second easyconfig, `NVHPC-20.7-GCC-9.3.0.eb` which does not contain the `CUDAcore` dependency and is fully _tuned_ to use the NVHPC-built-in CUDA?